docs/konflux-building-catalog.md

```diff
@@ -0,0 +1,42 @@
+## Building a File-Based Catalog from catalog template
+
+The `fbc-builder` pipeline in Konflux now allows you to generate the `catalog.json` file directly,  
+simplifying the process of creating a file-based catalog.  
+The `run-opm-command` task uses the `opm` command to build the catalog from the `catalog-template.yaml`.
+
+### Generating the `catalog.json` with the `fbc-builder` Pipeline
+
+To generate the `catalog.json` file, you need to run the `fbc-builder` pipeline.  
+For the `run-opm-command` task, you must set the following parameters: `OPM_ARGS`, `OPM_OUTPUT_PATH`, and `IDMS_PATH`.
+
+**Example:**
+```yaml
+- name: OPM_ARGS
+  value:
+    - alpha
+    - render-template
+    - basic
+    - "v4.18/catalog-template.json"
+    - "--migrate-level=bundle-object-to-csv-metadata"
+- name: OPM_OUTPUT_PATH
+  value: "v4.18/catalog/gatekeeper-operator/catalog.json"
+- name: IDMS_PATH
+  value: ".tekton/images-mirror-set.yaml"
+```
+
+**Description of variables:**
+
+- `OPM_ARGS`: An array of arguments passed to `opm` when executed.
+- `OPM_OUTPUT_PATH`: The path where the output from `opm` will be stored (i.e., where `catalog.json` will be saved).
+  - Note: This task supports only the JSON format for catalog generation.
+- `IDMS_PATH`: The path to the ImageDigestMirrorSet file in YAML format.
+- `FILE_TO_UPDATE_PULLSPEC`: Relative path to a file (e.g., catalog-template.yml) in which pullspecs should be updated.
+
+Examples of complete `fbc-builder` pipeline configurations can be found in:
+
+- [`gatekeeper-fbc-v418-push.yaml`](../.tekton/gatekeeper-fbc-v418-push.yaml)  
+- [`gatekeeper-fbc-v418-pull-request.yaml`](../.tekton/gatekeeper-fbc-v418-pull-request.yaml)
```

Comment:
I am on the fence about whether we need those big yaml files. Couldn't simply share a snippet here on how the task can be integrated in the pipeline?

Reply:
Or do you think we can reuse the existing 4.13 files we already have in the repo?

Reply:
I think existing v4.13 sample should stay untouched. It is still valid approach for those who do not want to include run-opm-command task. 
We need those big yaml files to be sure everything is set correctly. The run-opm-command changed few `runAfter`, because it needed to be run before build. Therefore I am more inclined towards showing the whole `PipelineRun` as it is done for v4.13.

Reply:
I agree with this question. What I did in the past was link to specific PRs where I made the changes. If you want to use new/separate definitions then we should onboard a new component for 4.18 separately and then only modify the pipelines to add this task in the PR.